lib/models/movie.dart

```diff
@@ -1,36 +1,74 @@
 class Movie {
   final String title;
   final String originalTitle;
+  final String originalLanguage;
+  final int id;
+  final bool adult;
   final String releaseDate;
   final String overview;
   final double voteAverage;
-  final List<String> genres;
+  final List<int> genres;
   final String posterPath;
   final String backdropPath;
+  final double popularity;
+  final bool video;
+  final int voteCount;
 
-  const Movie(
-      {required this.title,
-      required this.originalTitle,
-      required this.releaseDate,
-      required this.overview,
-      required this.voteAverage,
-      required this.genres,
-      required this.posterPath,
-      required this.backdropPath});
+  const Movie({
+    required this.title,
+    required this.originalTitle,
+    required this.originalLanguage,
+    required this.id,
+    required this.adult,
+    required this.releaseDate,
+    required this.overview,
+    required this.voteAverage,
+    required this.genres,
+    required this.posterPath,
+    required this.backdropPath,
+    required this.popularity,
+    required this.video,
+    required this.voteCount,
+  });
 
   factory Movie.mockMovie() {
     return const Movie(
       title: 'The Super Mario Bros. Movie',
       originalTitle: 'The Super Mario Bros. Movie',
+      originalLanguage: 'en',
+      id: 1,
+      adult: false,
       releaseDate: '2023-04-05',
       overview: 'While working underground to fix a water main, '
           'Brooklyn plumbers—and brothers—Mario and Luigi are transported down '
           'a mysterious pipe and wander into a magical new world. But when the '
           'brothers are separated, Mario embarks on an epic quest to find Luigi.',
       voteAverage: 7.8,
-      genres: ['Animation', 'Family', 'Adventure', 'Fantasy', 'Comedy'],
+      genres: [],
       posterPath: 'assets/images/mario_movie_poster.jpg',
       backdropPath: 'assets/images/mario_movie_backdrop.jpg',
+      popularity: 200.7,
+      video: false,
+      voteCount: 400,
+    );
+  }
+
+  factory Movie.fromJson(Map<String, dynamic> json) {
+    return Movie(
+      title: json['title'],
+      originalTitle: json['original_title'],
+      originalLanguage: json['original_language'],
+      id: json['id'] as int,
+      adult: json['adult'] as bool,
+      releaseDate: json['release_date'],
+      overview: json['overview'] as String,
+      voteAverage: json['vote_average'].toDouble(),
+      genres: List<int>.from(json['genre_ids']),
+      posterPath: 'https://image.tmdb.org/t/p/w500${json['poster_path']}',
+      backdropPath: 'https://image.tmdb.org/t/p/w500${json['backdrop_path']}',
```

Comment:
We can move the concatenation of the base url to a getter method:
`String get posterUrl => 'https://image.tmdb.org/t/p/w500${posterPath}'`

Reply:
done